src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs

```diff
@@ -175,27 +198,34 @@ public static FormulaValue Text(EvalVisitor runner, EvalVisitorContext context,
                 }
             }
 
+            // We limit the format string size
+            if (formatString != null && formatString.Length > formatSize)
+            {
+                var customErrorMessage = StringResources.Get(TexlStrings.ErrTextFormatTooLarge, culture.Name);
+                return CommonErrors.CustomError(irContext, string.Format(customErrorMessage, formatSize));
+            }
+
             switch (args[0])
             {
                 case NumberValue num:
-                    resultString = num.Value.ToString(formatString ?? "g", culture);
-                    break;
-                case StringValue s:
-                    resultString = s.Value;
+                    if (TextFormatUtils.IsDateTimeFormat(formatString))
+                    {
+                        // It's a number, formatted as date/time. Let's convert it to a date/time value first
+                        resultString = ExpandDateTimeExcelFormatSpecifiers(formatString, "g", TextFormatUtils.NumberValueToDateTime(num), culture, runner.CancelationToken);
+                    }
+                    else
+                    {
+                        resultString = num.Value.ToString(formatString ?? "g", culture);
+                    }                    
                     break;
                 case DateValue d:
-                    formatString = ExpandDateTimeFormatSpecifiers(formatString, culture);
-                    resultString = d.Value.ToString(formatString ?? "M/d/yyyy", culture);
+                    resultString = ExpandDateTimeExcelFormatSpecifiers(formatString, "M/d/yyyy", d.Value, culture, runner.CancelationToken);
```

Comment:
>ExpandDateTimeExcelFormatSpecifiers [](http://example.com/codeflow?start=35&length=35)

We should also be able to format Date/Time values as numbers. For example, `Text(DateTime(2022, 12, 14, 18, 0, 0), "0.000")` should return `"44909.75"`.